docs/migration.md

```diff
@@ -0,0 +1,102 @@
+# Migration from Preview 2
+
+If you are customers of [Cloud Foundry on Azure Preview 2](https://azure.microsoft.com/en-us/blog/cloud-foundry-on-azure-preview-2-now-available/), you need to do the following migration.
+
+## Migration
+
+1. Update bosh.yml.
+
+  Replace **BOSH-FOR-AZURE-URL**, **BOSH-FOR-AZURE-SHA1**, **BOSH-AZURE-CPI-URL**, **BOSH-AZURE-CPI-SHA1**, **STEMCELL-FOR-AZURE-URL** and **STEMCELL-FOR-AZURE-SHA1** with the values in the example file [**bosh.yml**](https://github.com/Azure/azure-quickstart-templates/blob/master/bosh-setup/bosh.yml).
+
+  Click [**HERE**](./deploy-bosh-manually.md#configure_bosh_yml) to learn more.
+
+2. You need to upgrade Ruby version.
+
+  ```
+  sudo apt-get install -y ruby2.0 ruby2.0-dev
+
+  sudo rm /usr/bin/ruby /usr/bin/gem /usr/bin/irb /usr/bin/rdoc /usr/bin/erb
+  sudo ln -s /usr/bin/ruby2.0 /usr/bin/ruby
+  sudo ln -s /usr/bin/gem2.0 /usr/bin/gem
+  sudo ln -s /usr/bin/irb2.0 /usr/bin/irb
+  sudo ln -s /usr/bin/rdoc2.0 /usr/bin/rdoc
+  sudo ln -s /usr/bin/erb2.0 /usr/bin/erb
+  sudo gem update --system
+  sudo gem pristine --all
+
+  sudo gem install bosh_cli -v 1.3016.0 --no-ri --no-rdoc
+  ```
+
+  If this step is not done, you may hit the error [**#1**](#error_installing_bosh_cli) and [**#2**](#failed_to_render_job_templates).
+
+3. You need to create containers `bosh` and `stemcell` in your default storage account.
```

Comment:
Please do it.

Best Regards,
Abel HU

? 2015?11?6??22:35?Bin Xia <notifications@github.com<mailto:notifications@github.com>> ???


In docs/migration.md<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fcloudfoundry-incubator%2fbosh-azure-cpi-release%2fpull%2f38%23discussion_r44141774&data=01%7c01%7cabelch%40064d.mgd.microsoft.com%7c23991df3727c4653f3ec08d2e6b7865e%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=lnsK4%2fiRiyN5d7CZN%2fW2mfc%2fSgHF%2f41ns2hq1A33nW8%3d>:

> +
> +  sudo rm /usr/bin/ruby /usr/bin/gem /usr/bin/irb /usr/bin/rdoc /usr/bin/erb
> +  sudo ln -s /usr/bin/ruby2.0 /usr/bin/ruby
> +  sudo ln -s /usr/bin/gem2.0 /usr/bin/gem
> +  sudo ln -s /usr/bin/irb2.0 /usr/bin/irb
> +  sudo ln -s /usr/bin/rdoc2.0 /usr/bin/rdoc
> +  sudo ln -s /usr/bin/erb2.0 /usr/bin/erb
> +  sudo gem update --system
> +  sudo gem pristine --all
> +
> +  sudo gem install bosh_cli -v 1.3016.0 --no-ri --no-rdoc
> +  ```
> +
> +  If this step is not done, you may hit the error [**#1**](#error_installing_bosh_cli) and [**#2**](#failed_to_render_job_templates).
> +
> +3. You need to create containers `bosh` and `stemcell` in your default storage account.


We can remove "ContainerNotFound" if we assume that customers have deployed BOSH after deploying bosh-setup templates.

-
Reply to this email directly or view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fcloudfoundry-incubator%2fbosh-azure-cpi-release%2fpull%2f38%2ffiles%23r44141774&data=01%7c01%7cabelch%40064d.mgd.microsoft.com%7c23991df3727c4653f3ec08d2e6b7865e%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=1ML7JGZHdO20NspWeuuZUCsTXVNEvhUlkrId6umFgCE%3d>.